src/main/java/org/mamute/i18n/UTF8Control.java

```diff
@@ -0,0 +1,86 @@
+package org.mamute.i18n;
+/* https://gist.github.com/Hasacz89/d93955ec91afc73a06e3 */
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Locale;
+import java.util.PropertyResourceBundle;
+import java.util.ResourceBundle;
+
+public class UTF8Control extends ResourceBundle.Control {
+    public ResourceBundle newBundle
+            (String baseName, Locale locale, String format, ClassLoader loader, boolean reload)
+            throws IllegalAccessException, InstantiationException, IOException {
+        // The below is a copy of the default implementation.
+        String bundleName = toBundleName(baseName, locale);
+        ResourceBundle bundle = null;
+        switch (format) {
+            case "java.class":
+                try {
+                    @SuppressWarnings("unchecked")
```

Comment:
This code is quite complex also, maybe we should extract two classes here, something like: JavaClassResourceLoader and JavaPropertiesResourceLoader, what do you think?